DllImportGenerator/DllImportGenerator/ManualTypeMarshallingHelper.cs

```diff
@@ -0,0 +1,65 @@
+#nullable enable
+
+using System.Linq;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.Interop
+{
+    static class ManualTypeMarshallingHelper
+    {
+        public const string ValuePropertyName = "Value";
+        public const string StackBufferSizeFieldName = "StackBufferSize";
+        public const string ToManagedMethodName = "ToManaged";
+        public const string FreeNativeMethodName = "FreeNative";
+
+        public static bool HasToManagedMethod(ITypeSymbol nativeType, ITypeSymbol managedType)
+        {
+            return nativeType.GetMembers(ToManagedMethodName)
+                    .OfType<IMethodSymbol>()
+                    .Any(m => m.Parameters.IsEmpty &&
```

Comment:
@elinor-fung Long ago had an argument I liked for placing the logic operator on the left side. I do find it clearer. Thoughts?

```CSharp
.Any(m => m.Parameters.IsEmpty
    && !m.ReturnsByRef
    && !m.ReturnsByRefReadonly
    && SymbolEqualityComparer.Default.Equals(m.ReturnType, managedType)
    && !m.IsStatic);
```

Reply:
I'm down for either. I didn't know if we had a pattern, so I just picked one. I'll switch them over in my next iteration.